### PR TITLE
ZEN-463: image from container updates

### PR DESCRIPTION
**Pull request by daniel112**
**Ticket**: [issue_id](https://jira.simpleviewtools.com/browse/ZEN-532)
> Depends on this keg-cli PR: https://github.com/simpleviewinc/keg-cli/pull/31

## Context

* Because of the new `image from updates`, we need to update the contents in the `/containers` to work with the new workflow

## Goal
* to ensure you can still run the same commands when working in a container
* Updating the containers folder to work with the new image workflow

## Updates

* `/containers/*`
   * followed a similar setup from the updated `evf` containers folder

### Setup
* Pull this branch locally
* ensure your `keg-cli` is on branch `ZEN-463-push-pull-tasks`
* ensure you don't have any keg consumer images locally

### Test 1
* run `keg consumer start`
   * it should pull from `ghcr.io/simpleviewinc/consumer:master`
* navigate to http://consumer-container-update.local.kegdev.xyz/
   * ensure the app loads

### Test 2
* make changes to your local `tap-events-force` or `keg-components` repo
   * *I added a console log and hard coded a text on  default button*
* open a new terminal tab
* run `keg consumer sync evf:install:start,components:install:start`
   * it should install and sync local evf && keg components
* Ensure the changes are reflected
    * [image](https://user-images.githubusercontent.com/3317835/105891225-9ce32b80-5fcd-11eb-8db7-edaa8bd87f01.png)

### CODE REVIEW

File: container/mutagen.yml [hunk 1]
```diff
@@ -30,10 +30,11 @@ actions:
       location: /keg/tap
       privileged: true
       cmds:
-        - yarn roll:dev
+        - yarn build
```
Comment (mpcarolin) [inline container/mutagen.yml:31]: what is the purpose of this change?

File: container/mutagen.yml [hunk 2]
```diff
@@ -30,10 +30,11 @@ actions:
...
     start:
       location: /keg/tap
       privileged: true
```
Comment (lancetipton) [inline container/mutagen.yml]: Not super important, but `privileged` is now on by default, and you have to explicitly turn it off. So you can remove the `privileged: true` if you want

File: container/values.yml [hunk 3]
```diff
@@ -9,21 +11,22 @@ env:
 
   # --- KEG-CLI ENV CONTEXT --- #
 
-  # Set the paths to the linked external app
-  # The app should be linked to the keg-cli with `kee`
-  # Example command to link the app => `keg tap link kee`
+  # Docker / Docker Compose paths
   KEG_DOCKER_FILE: "{{ cli.taps.links.consumer }}/container/Dockerfile"
   KEG_VALUES_FILE: "{{ cli.taps.links.consumer }}/container/values.yml"
+
```
Comment (lancetipton) [inline container/values.yml:15]: Not something you did, but all these `cli.taps.links.consumer` should be updated to `cli.taps.consumer.path` 

The globalConfig was updated a little bit ago, and I think this repo was forgotten about. 
Can you update these so they point to the correct locations. For example =>

WRONG
`KEG_DOCKER_FILE: "{{ cli.taps.links.consumer }}/container/Dockerfile"`

CORRECT
`KEG_DOCKER_FILE: "{{ cli.taps.consumer.path }}/container/Dockerfile"`

File: container/values.yml [hunk 4]
```diff
@@ -48,10 +51,14 @@ env:
   IMAGE: keg-test-consumer
   CONTAINER_NAME: keg-test-consumer
   VERSION: "0.0.1"
+  CHOKIDAR_USEPOLLING: 1
```
Comment (lancetipton) [inline container/values.yml]: `CHOKIDAR_USEPOLLING` is only needed for Taps, that use `keg-core` with webpack
Since the consumer doesn't, and the evf tap uses rollup in this context, you don't need to add it

**Review by lancetipton — CHANGES_REQUESTED** [review]
There are a few minor things, but over all looks really good.
Some of the items existed before your changes, but still need to be updated.
Let me know if you have questions. Thanks

**Review by mpcarolin — APPROVED** [review]
Worked for me 👍🏼

File: container/mutagen.yml (hunk 2 above)
Comment (daniel112) [reply container/mutagen.yml]: ok. updated

File: container/values.yml [hunk 5]
```diff
@@ -1,3 +1,5 @@
+start: /bin/bash /keg/keg-cli/containers/tap/run.sh
```
Comment (lancetipton) [inline container/values.yml]: This is not needed and should be removed.
A while ago, I started to migrate the sync actions into the `values.yml` file, but I never finished
Which is why you see this in other values files. I've since removed it, as seen [here](https://github.com/simpleviewinc/tap-events-force/blob/3bec8183f8404cd9f7e5f14e9474ae2dea0d6d0d/container/values.yml#L1) for the evf's image-from PR


I'd still like to do that migration at some point, but it not sure when.

**Review by lancetipton — CHANGES_REQUESTED** [review]
Added another comment for a thing  I found

File: container/values.yml (hunk 3 above)
Comment (daniel112) [reply container/values.yml:15]: updated

File: container/values.yml (hunk 4 above)
Comment (daniel112) [reply container/values.yml]: ohh ok got it. updated.
removed from `docker-compose.yml` and `values.yml`

File: container/mutagen.yml (hunk 1 above)
Comment (daniel112) [reply container/mutagen.yml:31]: `roll:dev` doesnt exist as a cmd anymore in tap-evf

File: container/values.yml (hunk 5 above)
Comment (daniel112) [reply container/values.yml]: updated

File: container/Dockerfile [hunk 6]
```diff
@@ -1,43 +1,23 @@
...
+ARG KEG_COPY_LOCAL
+
+# Add a FROM for the tap-base image to we can copy content from it
+ARG KEG_BASE_IMAGE=ghcr.io/simpleviewinc/tap:master
+FROM $KEG_BASE_IMAGE as tap-base
+ARG DOC_APP_PATH=/keg/app
+ENV DOC_APP_PATH=$DOC_APP_PATH
+
+# link repo to provider
+LABEL org.opencontainers.image.source $GIT_APP_URL
```
Comment (lancetipton) [inline container/Dockerfile:15]: What's this all about?

File: container/mutagen.yml (hunk 1 above)
Comment (lancetipton) [reply container/mutagen.yml:31]: Yah, I originally added this as a test cmd, way back when we were working on the rollup config. It should have been removed after, but instead started being used. So this is better.

**Review by lancetipton — APPROVED** [review]
Changes look good. Just hold on merging until this is merged => https://github.com/simpleviewinc/keg-cli/pull/31

Thanks

File: container/Dockerfile (hunk 6 above)
Comment (daniel112) [reply container/Dockerfile:15]: oh i was looking at the package from the ghcr.io and saw this
image

it just links the repo to the package provider. 
https://github.com/orgs/simpleviewinc/packages/container/package/consumer

Comment (lancetipton) [reply container/Dockerfile:15]: Nice, I didn't know that. I like it
